src/tools/kata-ctl/src/ops/check_ops.rs

```diff
@@ -1,44 +1,198 @@
 use crate::ops::version;
 
 use crate::args::{CheckArgument, CheckSubCommand, IptablesCommand, MetricsCommand};
 
-use anyhow::Result;
+use crate::check::{check_version,run_network_checks};
+
+use anyhow::{anyhow, Result};
+
+use std::fmt::{self,Formatter,Display};
 
 const NAME: &str = "kata-ctl";
 
+#[derive(Debug, Clone, Copy)]
+enum ArchType {
+    x86_64,
+    s390x,
+    aarch64,
+    powerpc64le,
+}
+
+enum UserType {
+    Root,
+    NonRoot,
+}
+
+enum PermissionType {
+    Privileged,
+    NonPrivileged,
+}
+
+type BuiltinCmdFp = fn(args: &str) -> Result<()>;
+
+
+struct CheckItem {
+    name: &'static str,
+    descr: &'static str,
+    fp: BuiltinCmdFp,
+    ut: UserType,
+    arch: ArchType,
+    perm: PermissionType,
+}
+
+static CHECK_LIST: &[CheckItem] = &[
+    CheckItem {
+        name: "check-cpu",
+        descr: "This parameter performs the cpu check",
+        fp: check,
+        ut: UserType::NonRoot,
+        arch: ArchType::x86_64,
+        perm: PermissionType::NonPrivileged,
+    },
+
+    CheckItem {
+        name: "check-network",
+        descr: "This parameter performs network check",
+        fp: run_network_checks,
+        ut: UserType::NonRoot,
+        arch: ArchType::x86_64,
+        perm: PermissionType::NonPrivileged,
+    },
+
+    CheckItem {
+        name: "check-version",
+        descr: "Retrieves latest kata version release",
+        fp: check_version,
+        ut: UserType::NonRoot,
+        arch: ArchType::x86_64,
+        perm: PermissionType::NonPrivileged,
+    },
+
+    CheckItem {
+        name: "check-cpu",
+        descr: "This parameter performs the cpu check",
+        fp: check,
+        ut: UserType::NonRoot,
+        arch: ArchType::s390x,
+        perm: PermissionType::NonPrivileged,
+    },
+
+    CheckItem {
+        name: "check-cpu",
+        descr: "This parameter performs the cpu check",
+        fp: check,
+        ut: UserType::NonRoot,
+        arch: ArchType::aarch64,
+        perm: PermissionType::NonPrivileged,
+    },
+
+    CheckItem {
+        name: "check-cpu",
+        descr: "This parameter performs the cpu check",
+        fp: check,
+        ut: UserType::NonRoot,
+        arch: ArchType::powerpc64le,
+        perm: PermissionType::NonPrivileged,
+    },
+
+];
+
+impl Display for ArchType {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+
+        match self {
+            ArchType::aarch64 => write!(f, "aarch64"),
+            ArchType::powerpc64le => write!(f, "powerpc64le"),
+            ArchType::s390x => write!(f, "s390x"),
+            ArchType::x86_64 => write!(f, "x86_64"),
+        }
+    }
+}
+
+// This function retrieves the cmd function passes as argument
+fn get_builtin_check_func(name: &str) -> Result<BuiltinCmdFp> {
+    for check in CHECK_LIST {
+        if check.name.eq(name) {
+            return Ok(check.fp);
+        }
+    }
+
+    Err(anyhow!("Invalid command: {:?}", name))
+}
+
+// This function is called from each CLI argument section
```

Comment:
Since the return value is a tuple, I think you may need to add what the `Result` and `bool` actually stands for in the comment.

Reply:
Thank you @Ji-Xinyou, I opt to only use Result as return value, because the bool returned value is not really useful.